pkg/agent/packetcapture/packetcapture_controller_linux.go

```diff
@@ -0,0 +1,884 @@
+	crdlisters "antrea.io/antrea/pkg/client/listers/crd/v1alpha1"
+	binding "antrea.io/antrea/pkg/ovs/openflow"
+	"antrea.io/antrea/pkg/util/env"
+	"antrea.io/antrea/pkg/util/ftp"
+)
+
+type storageProtocolType string
+
+const (
+	sftpProtocol storageProtocolType = "sftp"
+)
+
+const (
+	controllerName               = "PacketCaptureController"
+	resyncPeriod   time.Duration = 0
+
+	minRetryDelay = 5 * time.Second
+	maxRetryDelay = 300 * time.Second
+
+	defaultWorkers = 2
+
+	// defines how many capture request we can handle concurrently.
+	maxConcurrentCaptures = 4
+	// waiting captures will be marked as Pending unitl it can be processed.
+	maxWaitingCaptures = 32
+
+	// reason for timeout
+	captureTimeoutReason   = "PacketCapture timeout"
+	defaultTimeoutDuration = time.Second * time.Duration(60)
+
+	captureStatusUpdatePeriod = 10 * time.Second
+
+	// PacketCapture uses a dedicated Secret object to store authentication information for a file server.
+	// #nosec G101
+	fileServerAuthSecretName = "antrea-packetcapture-fileserver-auth"
+
+	// Max packet size for pcap capture.
+	maxSnapshotBytes = 65536
+)
+
+type packetCapturePhase string
+
+const (
+	packetCapturePhaseUnknown   packetCapturePhase = ""
+	packetCapturePhasePending   packetCapturePhase = "Pending"
+	packetCapturePhaseRunning   packetCapturePhase = "Running"
+	packetCapturePhaseCompleted packetCapturePhase = "Completed"
+)
+
+var (
+	packetDirectory = filepath.Join(os.TempDir(), "antrea", "packetcapture", "packets")
+	defaultFS       = afero.NewOsFs()
+)
+
+type packetCaptureState struct {
+	// name is the PacketCapture name.
+	name string
+	// capturedPacketsNum record how many packets have been captured. Due to the RateLimiter,
+	// this maybe not be the real-time data.
+	capturedPacketsNum int32
+	// targetCapturedPacketsNum is target number limit for our capture. When numCapturedPackets == targetCapturedPacketsNum, it means that
+	// the PacketCapture is done successfully.
+	targetCapturedPacketsNum int32
+	// updateRateLimiter controls the frequency of the updates to PacketCapture status.
+	updateRateLimiter *rate.Limiter
+	// pcapngFile is the file object for the packet file.
+	pcapngFile afero.File
+	// pcapngWriter is the writer for the packet file.
+	pcapngWriter *pcapgo.NgWriter
+}
+
+type Controller struct {
+	kubeClient            clientset.Interface
+	crdClient             clientsetversioned.Interface
+	packetCaptureInformer crdinformers.PacketCaptureInformer
+	packetCaptureLister   crdlisters.PacketCaptureLister
+	packetCaptureSynced   cache.InformerSynced
+	interfaceStore        interfacestore.InterfaceStore
+	nodeConfig            *config.NodeConfig
+	queue                 workqueue.TypedRateLimitingInterface[string]
+	sftpUploader          ftp.Uploader
+	newPacketSourceFn     func(c *Controller, pc *crdv1alpha1.PacketCapture) (PacketSource, error)
+	cond                  *sync.Cond
+	// a name-phase mapping for all capture instances.
```

Comment:
Make sure the comment starts with capitalized word.
```suggestion
	// A name-phase mapping for all PacketCapture CRs.
```